### DISCUSSION TITLE
Case sensitive filtering

### DISCUSSION THREAD

**Discussion by herlon214** [meilisearch/product#555] [Feedback & Feature Proposal]
Hello, I've been using meilisearch and it's been awesome!

I have a search which filters by ID and these if I specify the filter `id = 'K8vZ9179yTf'`  it also gives me results for `K8vZ9179YTf` which is clearly a case insensitive search.

Is there a way to enable the filter with case sensitive? For my case giving those two different results in the search really makes the results very wrong.

**Comment by oluademola** [comment]
Hello @herlon214 

It is our pleasure to hear about your experience, and thank you very much for the feedback! Unfortunately, this is indeed an expected behavior with the current search engine, because filtering on string values is case-insensitive, as seen in the [dedicated documentation](https://docs.meilisearch.com/learn/advanced/filtering_and_faceted_search.html#conditions). However, we'd be more than happy if you could submit this as a feature request by starting a new discussion on the [product repo](https://github.com/meilisearch/product/discussions/) explaining your use case. This would be very helpful to the product team. They would evaluate this and could be considered for design changes in a future release. 

Hope this helps. Thank you.

**Reply by herlon214** [reply]
Thanks for the reply, I did a workaround the ids by converting them to int using crc32 and it's working fine for me now. Maybe someone else can take the case-sensitive idea further.

**Comment by gmourier** [comment]
Hey @herlon214 and @oluademola 👋

I've moved this discussion to the product side. Thanks for your feedback and the provided work-around! 

Relates to https://github.com/meilisearch/product/discussions/418

**Comment by CaroFG** [comment]
Adding [Discord comments](https://discord.com/channels/1006923006964154428/1144830028270547064/1145711755834687508) about the feature:
<img width="1220" alt="@CaroFG thank you for your reply
meilisearch has been great but if we can have a settings which can have turn on/off string case-sensitiveness filtering would be awesome
in my case is to filter the client’ conpany name or short name which should be critical since “IpL” and “IPL” could be different company" src="https://github.com/meilisearch/product/assets/48251481/a22a66c9-2d61-436c-bc3d-cbf41b33f985">

**Comment by CaroFG** [comment]
[Another one](https://discord.com/channels/1006923006964154428/1144830028270547064/1216961944460857386)
<img width="1181" alt="I would love this as well. In my use-case, I am trying to filter based on database keys. I'm storing my data in firebase rtd but want to use meilisearch for my complex search queries." src="https://github.com/meilisearch/product/assets/48251481/51c19bca-7905-48b0-a957-419c885f3347">
